.github/workflows/pull_request_template.md

```diff
@@ -0,0 +1,14 @@
+## Describe your changes
+
+
+## Describe any testing you have performed
+
+## Link relevant issues
+
+## Checklist when you are ready to request a review
+
+- [ ] I have performed a self-review of my code
+- [ ] I have added docstrings
+- [ ] CHANGELOG.md has been updated with any functionality change
+- [ ] Request review from all relevant developers
+- [ ] Change pull request label to 'Waiting for review' 
```

Comment:
I think we need the declaration that the contribution is Apache 2.

Reply:
we also need to update file headers etc to be Apache2

Reply:
updated check list